Tests/ValTests/CXXTests.swift

```diff
@@ -14,6 +18,15 @@ func check(_ haystack: String, contains needle: String.SubSequence) {
     """)
 }
 
+/// Remove trailing newlines from the given string subsequence
```

Comment:
```suggestion
/// Removes trailing newlines from the given string subsequence.
func removingTrailingNewlines(_ s: Substring) -> Substring {
```

- I'm suggesting `removingTrailingNewlines` because We've been trying to avoid abbreviated names for functions.
- I'm suggesting `Substring` rather than `String.SubSequence` because I think it's more straightforward.

Note that I would typically implement this function as a method of `StringProtocol` so that it can apply to both strings and substrings without polluting the global namespace:
```
extension StringProtocol {

  func removingTrailingNewlines() -> Self.SubSequence {
    if let i = self.lastIndex(where: { !$0.isNewline }) {
      return self.prefix(through: i)
    } else {
      return self.prefix(upTo: startIndex)
    }
  }

}
```